sdks/java/java8tests/src/test/java/org/apache/beam/sdk/transforms/CombineJava8Test.java

```diff
@@ -131,4 +140,31 @@ public void testCombinePerKeyInstanceMethodReference() {
         KV.of("c", 4));
     pipeline.run();
   }
+
+  /**
+   * Tests that we can serialize Combine fns constructed from a lambda. Lambdas can be problematic
+   * because the {@link Class} object is synthetic and cannot be deserialized.
+   */
+  @Test
+  public void testLambdaSerialization() {
+    SerializableFunction<Iterable<Object>, Object> combiner = xs -> Iterables.getFirst(xs, 0);
+
+    try {
+      SerializableUtils.clone(combiner.getClass());
+      Assert.fail("Expected lambda class serialization to fail. "
```

Comment:
Not sure we want to fail here. Maybe just log this to stderr? Specifically, would we really want to *break the build* just because a new version of Java is able to serialize it?